### PR TITLE
Add block recall from tier2

### PR DESCRIPTION
**Pull request by tangledbytes**
### Explain the changes
This PR adds block recall feature from tier2 filesystem. It does not implement retries and rather would let the S3 client to retry the reads.

- [ ] Doc added/updated
- [ ] Tests added

### CODE REVIEW

File: src/native/fs/fs_napi.cpp [hunk 1]
```diff
@@ -578,7 +578,19 @@ struct Stat : public FSWorker
         if ((int)info.Length() == 3) {
             Napi::Object options = info[2].As<Napi::Object>();
             if (options.Has("use_lstat")) _use_lstat = options.Get("use_lstat").ToBoolean();
-            if (options.Has("skip_user_xattr")) _skip_user_xattr = options.Get("skip_user_xattr").ToBoolean();
+            if (options.Has("skip_user_xattr") && options.Get("skip_user_xattr").ToBoolean()) {
```
Comment (utkarsh-pro) [inline src/native/fs/fs_napi.cpp:581]: Add `if` `else`.

File: src/native/fs/fs_napi.cpp [hunk 2]
```diff
@@ -980,7 +992,19 @@ struct Readfile : public FSWorker
         if ((int)info.Length() == 3) {
             Napi::Object options = info[2].As<Napi::Object>();
             if (options.Has("read_xattr")) _read_xattr = options.Get("read_xattr").ToBoolean();
-            if (options.Has("skip_user_xattr")) _skip_user_xattr = options.Get("skip_user_xattr").ToBoolean();
+            if (options.Has("skip_user_xattr") && options.Get("skip_user_xattr").ToBoolean()) {
```
Comment (utkarsh-pro) [inline src/native/fs/fs_napi.cpp:995]: Eliminate duplication.

File: src/agent/block_store_services/block_store_fs.js [hunk 3]
```diff
@@ -85,6 +85,26 @@ class BlockStoreFs extends BlockStoreBase {
...
+            const stat = await nb_native().fs.stat(fs_context, block_path, {
+              xattr_get_keys: [config.BLOCK_STORE_FS_XATTR_QUERY_MIGSTAT]
+            });
+
+            const migstat = JSON.parse(stat.xattr[config.BLOCK_STORE_FS_XATTR_QUERY_MIGSTAT] || '{}');
+
+            if (migstat.State === 'MIGRATED') {
+                // if not yet trying to premigrate, try now.
+                if (migstat.TargetState !== 'PREMIGRATED') {
+                    const file = await nb_native().fs.open(fs_context, block_path);
```
Comment (utkarsh-pro) [inline src/agent/block_store_services/block_store_fs.js:98]: Move `open` to top and move the whole block inside try catch.

**Comment by utkarsh-pro** [comment]
Opened an issue identifying the gap in this PR.

**Review by guymguym — APPROVED** [review]
🤩

**Pull request merged**